### PR TITLE
i3s: support texture atlas

### PR DESCRIPTION
**Pull request by belom88**
I implemented support of the texture atlas according to https://github.com/Esri/i3s-spec/blob/master/docs/1.7/geometryUVRegion.cmn.md . But it doesn't work for some reason.
Screenshot from 2021-03-19 11-46-28

This PR works for San Francisco 1.7:
image
image

### CODE REVIEW

**Review by ibgreen — COMMENTED** [review]
- I'd recommend setting up some test cases for the uvRegion array on CPU so that it looks good.
- Have you confirmed that the define is activated? You can put a breakpoint in the Shader contstructor in shader.js and look at the generated source code.


Does normalizeAttribute convert the attribute to Float32? This is not necessary, the GPU can normalize if we set up the attribute correctly. We can look at that in another PR.

**Comment by belom88** [comment]
> Have you confirmed that the define is activated?

Yes

> Does normalizeAttribute convert the attribute to Float32?

Yes. I just found `normalized` option and use it in this PR. It also works for colors.

File: examples/experimental/i3s-17-and-debug/mesh-layer/mesh-layer.js [hunk 1]
```diff
@@ -107,12 +107,17 @@ export default class MeshLayer extends SimpleMeshLayer {
 
     const shaders = this.getShaders();
 
+    const customDefines = {};
+    if (mesh.attributes.uvRegions) {
```
Comment (ibgreen) [inline examples/experimental/i3s-17-and-debug/mesh-layer/mesh-layer.js:111]: I am not super fond of using defines. I think the overhead of always using a region is small, we could just use a default constant region attribute instead.

File: modules/i3s/src/lib/parsers/parse-i3s-tile-content.js [hunk 2]
```diff
@@ -146,8 +148,9 @@ async function parseI3SNodeGeometry(arrayBuffer, tile = {}, options) {
   content.attributes = {
     positions: attributes.position,
     normals: attributes.normal,
-    colors: normalizeColors(attributes.color),
+    colors: normalizeAttribute(attributes.color), // Normalize from UInt8
     texCoords: attributes.uv0,
+    uvRegions: normalizeAttribute(attributes.uvRegion), // Normalize from UInt16
```
Comment (ibgreen) [inline modules/i3s/src/lib/parsers/parse-i3s-tile-content.js:153]: You could just use the luma.gl feature to set this to a constant value (single array) when no region is available.

**Review by ibgreen — APPROVED** [review]

**Pull request merged**